### PR TITLE
Add support of Apache tomEE (1.7.2 +)

### PR DESCRIPTION
**Pull request by tcollignon**
HI, 

This pull request allow to use tomEE(http://tomee.apache.org/index.html) with webby (because support of tomEE was added in Cargo before).

Warning : this can work with only 1.7.2 and upper version of tomEE

### CODE REVIEW

File: org.sonatype.m2e.webby/src/org/sonatype/m2e/webby/internal/config/WarConfigurationExtractor.java [hunk 1]
```diff
@@ -198,8 +198,7 @@ public WarConfiguration getConfiguration(IMavenProjectFacade mvnFacade, MavenPro
   private String resolve(String basedir, String path) {
     String result = path;
     if(path != null && basedir != null) {
-      path = path.replace('\\', File.separatorChar).replace('/', File.separatorChar);
-
+      path = path.replace('\\', '/');
```
Comment (rkrzewski) [inline org.sonatype.m2e.webby/src/org/sonatype/m2e/webby/internal/config/WarConfigurationExtractor.java]: Why is this change necessary? The old code appears correct for both Windows and Unix filesystems.

Comment (tcollignon) [inline org.sonatype.m2e.webby/src/org/sonatype/m2e/webby/internal/config/WarConfigurationExtractor.java]: In fact the old code don't work in windows with tomee in war extra classpath, because we have some "\" character
I think "/" it's better to work with every OS
=> I Have test this on Windows and Mac Os X

Comment (rkrzewski) [inline org.sonatype.m2e.webby/src/org/sonatype/m2e/webby/internal/config/WarConfigurationExtractor.java]: Who do you mean by "we"? Will this change work for other users of Webby?
What about Tomcat and Jetty, is this change compatible with them, especially on Windows?
AFAIK, "/" needs to be used in URLs, but File.separatorChar needs to be used in file system paths to be portable.

**Comment by tcollignon** [comment]
Ok, I have changed the line in WarConfigurationExtractor and back to original effect, and I have refactor some code to use same method every where. I have added a test too.
I have test on windows and Mac os X, and it's ok.
tell me if it's ok for you now.
Thanks

File: org.sonatype.m2e.webby/src/org/sonatype/m2e/webby/internal/util/PathSelector.java [hunk 2]
```diff
@@ -57,9 +57,8 @@ private static String normalizePattern(String pattern) {
     return normalized;
   }
 
-  private static String normalizePath(String path) {
-    String normalized = path.replace((File.separatorChar == '/') ? '\\' : '/', File.separatorChar);
-    return normalized;
+  public static String normalizePath(String path) {
```
Comment (rkrzewski) [inline org.sonatype.m2e.webby/src/org/sonatype/m2e/webby/internal/util/PathSelector.java]: Great, I like this approach much better! I'd like to suggest one more change though:
```java
return path.replaceAll("[\\/]", File.separatorChar);
```
It does the same thing, but is a little bit more compact.

Comment (tcollignon) [inline org.sonatype.m2e.webby/src/org/sonatype/m2e/webby/internal/util/PathSelector.java]: I have try this on windows and my test WarConfigurationExtractorTest is in error : 
````
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: 1
	at java.lang.String.charAt(Unknown Source)
	at java.util.regex.Matcher.appendReplacement(Unknown Source)
	at java.util.regex.Matcher.replaceAll(Unknown Source)
	at java.lang.String.replaceAll(Unknown Source)
	at org.sonatype.m2e.webby.internal.util.PathSelector.normalizePath(PathSelector.java:61)
	at org.sonatype.m2e.webby.internal.config.WarConfigurationExtractor.resolve(WarConfigurationExtractor.java:202)
 ````

**Comment by tcollignon** [comment]
According to String javadoc, we should add Matcher.quoteReplacement if we have a "\"
Note that backslashes (\) and dollar signs ($) in the replacement string may cause the results to be different than if it were being treated as a literal replacement string; see Matcher.replaceAll. Use java.util.regex.Matcher.quoteReplacement to suppress the special meaning of these characters, if desired.

**Comment by rkrzewski** [comment]
Right, that was the reason the test was failing on Windows. All is clear now.

**Comment by rkrzewski** [comment]
Thank you for your contribution, @tcolligon!
